### PR TITLE
US889115: Initial implementation

### PR DESCRIPTION
**Pull request by kusumaghoshdastidar**
https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=889115

### CODE REVIEW

**Review by dermot-hardy — CHANGES_REQUESTED** [review]
- `help` goal required
- registry name parsing
- docker inspect for checking image presence
- maven http config
- another ticket for creds store support

**Comment by kusumaghoshdastidar** [comment]
> * `help` goal required
> * registry name parsing
> * docker inspect for checking image presence
> * maven http config
> * another ticket for creds store support

Ticket to add Credential Store support: [895059](https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=895059)

File: docker-versions-maven-plugin/pom.xml [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-plugin-plugin</artifactId>
+                <configuration>
+                    <goalPrefix>docker-versions</goalPrefix>
+                    <skipErrorNoDescriptorsFound>true</skipErrorNoDescriptorsFound>
```
Comment (dermot-hardy) [inline docker-versions-maven-plugin/pom.xml:91]: Why has this `skipErrorNoDescriptorsFound` been added?

File: docker-versions-maven-plugin/pom.xml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+                </configuration>
+                <executions>
+                    <execution>
+                        <id>mojo-descriptor</id>
+                        <goals>
+                            <goal>descriptor</goal>
+                        </goals>
+                    </execution>
```
Comment (dermot-hardy) [inline docker-versions-maven-plugin/pom.xml:99]: Isn't this a default binding for `maven-plugin` packaging?  Won't adding it here just cause it to run twice?

**Review by dermot-hardy — CHANGES_REQUESTED** [review]

File: docker-versions-maven-plugin/pom.xml (hunk 2 above)
Comment (kusumaghoshdastidar) [reply docker-versions-maven-plugin/pom.xml]: Yes, it did run twice. I didn't realize there was a default.

**Review by dermot-hardy — APPROVED** [review]

**Pull request merged**